packages/core/src/sync-realtime/index.ts

```diff
@@ -395,9 +411,24 @@ export const createRealtimeSync = (
         }
 
         return isMatched;
-      }),
+      });
+
+      for (let i = 0; i < callTraceFilters.length; i++) {
+        if (isFilterMatched[i]!) matchedFilters.add(callTraceFilters[i]!);
```

Comment:
Could move this into the map above

Reply:
After a second glance, with the way there are two separate branches in the block above for filters that have a factory or not, I think it is more readable how it is right now